app/models/parser/twitter_item.rb

```diff
@@ -19,47 +19,63 @@ def patterns
     # Main function for class
     def parse_data_for_parser(_doc, _original_url, _jsonld_array)
       @url.gsub!(/(%23|#)!\//, '')
+      @url.gsub!(/\s/, '')
       @url = replace_subdomain_pattern(url)
+      
       parts = url.match(TWITTER_ITEM_URL)
       user, id = parts['user'], parts['id']
       
-      @parsed_data['raw']['api'] = {}
-      handle_twitter_exceptions do
-        @parsed_data['raw']['api'] = twitter_client.status(id, tweet_mode: 'extended').as_json
+      @parsed_data['raw']['api'] = tweet_lookup(id)
+
+      @parsed_data[:error] = parsed_data['raw']['api']['errors']
+
+      if @parsed_data[:error] 
+        title = url
+        description = ''
+        author_picture = ''
+        published_at = ''
+        html = ''
+        author_name = user
+        author_url = get_author_url(user)
+      elsif @parsed_data[:error].nil?
+        raw_data = parsed_data['raw']['api']['data'][0]
+        raw_user_data = parsed_data['raw']['api']['includes']['users'][0]
+
+        title = raw_data['text'].squish
+        description = title
+        author_picture = raw_user_data['profile_image_url'].gsub('_normal', '')
+        published_at = raw_data['created_at']
+        html = html_for_twitter_item(url)
+        author_name = raw_user_data['name']
+        author_url = get_author_url(user) || raw_user_data['url'] || RequestHelper.top_url(url)
       end
-      @parsed_data[:error] = parsed_data.dig(:raw, :api, :error)
+ 
       @parsed_data.merge!({
```

Comment:
I might suggest just setting the values on parsed_data in the if/else block above since there is so little data we want to set in the error case. if we need to set any default values (like picture), then consider doing it before the if/else block, since the merge! would overwrite them when called.